src/wallet/api/walletApi.ts

```diff
@@ -104,12 +92,18 @@ const walletApi = {
     }
   },
 
-  async getHistory(wallet: HathorWalletServiceWallet, token: string | null = null): Promise<HistoryResponseData> {
-    // TODO add pagination parameters
-    const data = { params: {} };
-    if (token) {
-      data['params']['token_id'] = token;
+  async getTokens(wallet: HathorWalletServiceWallet): Promise<TokensResponseData> {
+    const axios = await axiosInstance(wallet);
+    const response = await axios.get('wallet/tokens');
+    if (response.status === 200 && response.data.success === true) {
```

Comment:
Actually not. If we decide to resolve promises only for 2xx the code will be like that:

```
try{
    const response = await axios.get('wallet/addresses');
    if (response.data.success === true) {
      return response.data;
    } else {
      throw new WalletRequestError('Error getting wallet addresses.');
    }
catch (e) {
  throw new WalletRequestError(e.message);
}
```

The `if...else` would still be needed because of the `response.data.success`

Reply:
You're right. Thanks!